### ISSUE TITLE
[1.4.2-beta.5+] Cannot find module when babel-runtime listed in devDependencies

### ISSUE THREAD

**Issue by eldog** [meteor/meteor#7872]
As mentioned in the pull request #7668, the app crashed with `Error: Cannot find module '../core-js/symbol/iterator'`. Moving `babel-runtime` from `devDependencies` to `dependencies` fixed this.

The reason we have `babel-runtime` in our `package.json` is for other node tools we use (unit tests for small bits of code), and we tried to create the same runtime as meteor for them.

Repo with reproduction app:
 
https://github.com/eldog/meteor-dev-dependency-bug

**Comment by benjamn** [comment]
Thanks for taking the time to open this issue!

**Comment by thomasvanlankveld** [comment]
In case anyone else gets `Error: Cannot find module '../core-js/symbol/iterator'` after updating meteor to `1.4.2.3`, you may need to `meteor npm install babel-runtime`.

**Comment by abernix** [comment]
It will be added automatically in new projects, but for existing projects you should also add the `--save` flag to that command:

    meteor npm install babel-runtime --save

This is noted [in the changelog](https://github.com/meteor/meteor/blob/devel/History.md#v1421) for 1.4.2.1.

**Comment by ejfrancis** [comment]
I'm seeing this behavior when I'm using a local package with `METEOR_PACKAGE_DIRS=dir/`. If I put my local package into a `<project-root>/package/<package-name>` then local builds work fine, but if I put the package elsewhere and set `METEOR_PACKAGE_DIRS=../package-dir/` the error above happens.

To reproduce you can pull down my repo https://github.com/ejfrancis/prosemirror/ at commit 1465251a95c451a344a26e5bf0152bf1e5146a07 and running `meteor npm run test`. It will attempt to run two servers inside the demo app in `demo/` on different ports but never gets past the first one running. It will run something like this:

```
METEOR_PACKAGE_DIRS=/Users/efrancis/projects/ NODE_ENV=test  meteor --port 3005
```

I also get the error described in the now closed issue #5923 every 1 in 5 or so fresh builds

-------
### System
`meteor --version` output: 1.4.2.3
`meteor list` output:
```
accounts-base                       1.2.14  A user account system
blaze-html-templates                1.0.5  Compile HTML templates into reactive UI with Meteor Blaze
dburles:mongo-collection-instances  0.3.5  Access your Mongo instances
ecmascript                          0.6.1  Compiler plugin that supports ES2015+ in all .js files
es5-shim                            4.6.15  Shims and polyfills to improve ECMAScript 5 support
jquery                              1.11.10  Manipulate the DOM using CSS selectors
meteor-base                         1.0.4  Packages that every Meteor app needs
mobile-experience                   1.0.4  Packages for a great mobile user experience
mongo                               1.1.14  Adaptor for using MongoDB and Minimongo over DDP
prosemeteor:prosemirror             0.1.1+ Prosemirror integration for Meteor
reactive-var                        1.0.11  Reactive variable
shell-server                        0.2.1  Server-side component of the `meteor shell` command.
standard-minifier-css               1.3.2  Standard css minifier used with Meteor apps by default.
standard-minifier-js                1.2.1  Standard javascript minifiers used with Meteor apps by default.
tracker                             1.1.1  Dependency tracker to allow reactive callbacks
xolvio:backdoor                     0.2.1  Runs arbitrary code on server
```


OSX 10.11.6

**Comment by ejfrancis** [comment]
To expand upon this, the error says 

```
W20170113-17:59:46.941(-8)? (STDERR) Error: Cannot find module '../core-js/object/assign'

W20170113-17:59:46.941(-8)? (STDERR)     at require (packages/modules-runtime.js:109:19)

W20170113-17:59:46.942(-8)? (STDERR)     at meteorInstall.node_modules.meteor.prosemeteor:prosemirror.node_modules.babel-runtime.helpers.extends.js (/Users/efrancis/projects/prosemirror/demo/.meteor/local/build/programs/server/packages/prosemeteor_prosemirror.js:2510:15)

W20170113-17:59:46.942(-8)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:181:9)

W20170113-17:59:46.942(-8)? (STDERR)     at Module.require (packages/modules-runtime.js:106:16)

W20170113-17:59:46.942(-8)? (STDERR)     at Module.Mp.import (/Users/efrancis/.meteor/packages/modules/.0.7.7.yjn28g++os+web.browser+web.cordova/npm/node_modules/reify/lib/runtime.js:70:16)

W20170113-17:59:46.942(-8)? (STDERR)     at meteorInstall.node_modules.meteor.prosemeteor:prosemirror.lib.imports.api.documents.both.methods.js (/Users/efrancis/projects/prosemirror/demo/.meteor/local/build/programs/server/packages/prosemeteor_prosemirror.js:2170:207)

W20170113-17:59:46.942(-8)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:181:9)

W20170113-17:59:46.942(-8)? (STDERR)     at Module.require (packages/modules-runtime.js:106:16)

W20170113-17:59:46.942(-8)? (STDERR)     at Module.Mp.import (/Users/efrancis/.meteor/packages/modules/.0.7.7.yjn28g++os+web.browser+web.cordova/npm/node_modules/reify/lib/runtime.js:70:16)

W20170113-17:59:46.942(-8)? (STDERR)     at meteorInstall.node_modules.meteor.prosemeteor:prosemirror.lib.imports.api.authority.server.authority.js (/Users/efrancis/projects/prosemirror/demo/.meteor/local/build/programs/server/packages/prosemeteor_prosemirror.js:1424:829)
```

So `node_modules/babel-runtime/helpers/extends.js` is trying to import  `node_modules/babel-runtime/core-js/object/assign`, this fails even though the file is there. This is the contents of `node_modules/babel-runtime/core-js/object`

```
-rw-r--r--  1 efrancis    94 Jul 26 15:09 assign.js
-rw-r--r--  1 efrancis    94 Jul 26 15:09 create.js
-rw-r--r--  1 efrancis   105 Jul 26 15:09 define-properties.js
-rw-r--r--  1 efrancis   103 Jul 26 15:09 define-property.js
-rw-r--r--  1 efrancis    95 Jul 26 15:09 entries.js
-rw-r--r--  1 efrancis    94 Jul 26 15:09 freeze.js
-rw-r--r--  1 efrancis   115 Jul 26 15:09 get-own-property-descriptor.js
-rw-r--r--  1 efrancis   116 Jul 26 15:09 get-own-property-descriptors.js
-rw-r--r--  1 efrancis   110 Jul 26 15:09 get-own-property-names.js
-rw-r--r--  1 efrancis   112 Jul 26 15:09 get-own-property-symbols.js
-rw-r--r--  1 efrancis   104 Jul 26 15:09 get-prototype-of.js
-rw-r--r--  1 efrancis   101 Jul 26 15:09 is-extensible.js
-rw-r--r--  1 efrancis    97 Jul 26 15:09 is-frozen.js
-rw-r--r--  1 efrancis    97 Jul 26 15:09 is-sealed.js
-rw-r--r--  1 efrancis    90 Jul 26 15:09 is.js
-rw-r--r--  1 efrancis    92 Jul 26 15:09 keys.js
-rw-r--r--  1 efrancis   106 Jul 26 15:09 prevent-extensions.js
-rw-r--r--  1 efrancis    92 Jul 26 15:09 seal.js
-rw-r--r--  1 efrancis   104 Jul 26 15:09 set-prototype-of.js
-rw-r--r--  1 efrancis    94 Jul 26 15:09 values.js
```

I tried this on Ubuntu 15.04 and it happens as well
